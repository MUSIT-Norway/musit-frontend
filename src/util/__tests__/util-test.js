import assert from 'assert'
import { isDateBiggerThanToday, parseISODateNonStrict, parseISODateStrict } from '../'
import moment from 'moment'

describe('parseISODateNonStrict', () => {
  it('should accept full iso timestamp', () => {
    const date = '2016-09-07T22:56:00.000Z'
    const parsed = parseISODateNonStrict(date)
    expect(parsed).toMatchSnapshot()
  })

  it('should accept standard simple ISO date format', () => {
    const date = '2016-12-23'
    const parsed = parseISODateNonStrict(date)
    assert(parsed.isValid() === true)
  })
})

describe('parseISODateStrict', () => {
  it('should reject full iso timestamp', () => {
    const date = '2016-09-07T22:56:00+00:00'
    const parsed = parseISODateStrict(date)
    assert(parsed.isValid() === false)
  })

  it('should accept standard simple ISO date format', () => {
    const date = '2016-09-07'
    const parsed = parseISODateStrict(date)
    assert(parsed.isValid() === true)
  })
})

describe('isDateBiggerThanToday', () => {
  it('Check Today', () => {
    const newDate = moment()
    assert(isDateBiggerThanToday(newDate) === false)
  })
  it('Check one day plus Today', () => {
    const newDate = moment().add(1, 'd')
    assert(isDateBiggerThanToday(newDate) === true)
  })
  it('Check one month plus Today', () => {
    const newDate = moment().add(1, 'M')
    assert(isDateBiggerThanToday(newDate) === true)
  })
  it('Check one year plus Today', () => {
    const newDate = moment().add(1, 'y')
    assert(isDateBiggerThanToday(newDate) === true)
  })
  it('Check one day subtract from Today', () => {
    const newDate = moment().subtract(1, 'd')
    assert(isDateBiggerThanToday(newDate) === false)
  })
  it('Check one month subtract from Today', () => {
    const newDate = moment().subtract(2, 'M')
    assert(isDateBiggerThanToday(newDate) === false)
  })
  it('Check one year subtract from Today', () => {
    const newDate = moment().subtract(1, 'y')
    assert(isDateBiggerThanToday(newDate) === false)
  })
  it('Check Today minus one month three days', () => {
    let newDate = moment().subtract(3, 'd')
    newDate = newDate.subtract(1, 'd')
    assert(isDateBiggerThanToday(newDate) === false)
  })
  it('Check Today minus one month 15 days', () => {
    let newDate = moment().subtract(15, 'd')
    newDate = newDate.subtract(1, 'd')
    assert(isDateBiggerThanToday(newDate) === false)
  })
  it('Check Today minus one month 29 days', () => {
    let newDate = moment().subtract(29, 'd')
    newDate = newDate.subtract(1, 'd')
    assert(isDateBiggerThanToday(newDate) === false)
  })
  it('Check Today plus one month 14 days', () => {
    let newDate = moment().add(14, 'd')
    newDate = newDate.add(1, 'd')
    assert(isDateBiggerThanToday(newDate) === true)
  })
  it('Check Today plus one month 29 days', () => {
    let newDate = moment().add(29, 'd')
    newDate = newDate.add(1, 'd')
    assert(isDateBiggerThanToday(newDate) === true)
  })
  it('Check blank', () => {
    const newDate = ''
    assert(isDateBiggerThanToday(newDate) === false)
  })
  it('Check null', () => {
    const newDate = null
    assert(isDateBiggerThanToday(newDate) === false)
  })
  it('Check undefined', () => {
    const newDate = undefined
    assert(isDateBiggerThanToday(newDate) === false)
  })
})
