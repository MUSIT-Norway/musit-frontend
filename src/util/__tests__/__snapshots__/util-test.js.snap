exports[`SortObject Check blank 1`] = `"{}"`;

exports[`SortObject Check null 1`] = `"{}"`;

exports[`SortObject Check null array 1`] = `"[]"`;

exports[`SortObject Check undefined 1`] = `"{}"`;

exports[`SortObject Sort date value 1`] = `
Array [
  Object {},
  Object {},
  Object {
    "floatInStr": "2.2"
  },
  Object {
    "date": "1982-10-31T23:00:00.000Z",
    "float": 2.5,
    "floatInStr": "2.5",
    "int": 200,
    "intInStr": "211",
    "string": "ååå"
  },
  Object {
    "date": "2003-06-31T23:00:00.000Z",
    "float": 452.5,
    "floatInStr": "33232.5",
    "int": 190,
    "intInStr": "23333",
    "string": "a"
  },
  Object {
    "date": "2016-10-01T23:00:00.000Z",
    "float": 26.5,
    "floatInStr": "26.5",
    "int": 22,
    "intInStr": "222",
    "string": "ZZ"
  },
  Object {
    "date": "2016-10-11T23:00:00.000Z",
    "float": 12.5,
    "floatInStr": "13232.5",
    "int": 333,
    "intInStr": "11111",
    "string": "zz"
  },
  Object {
    "date": "2016-10-22T23:00:00.000Z",
    "float": 2.6,
    "floatInStr": "2.8",
    "int": 0,
    "intInStr": "0",
    "string": ""
  },
  Object {
    "date": "2016-11-31T23:00:00.000Z",
    "float": 222.5,
    "floatInStr": "22332.5",
    "int": 190,
    "intInStr": "23",
    "string": "a"
  }
]
`;

exports[`SortObject Sort date value descending 1`] = `
Array [
  Object {
    "date": "2016-11-31T23:00:00.000Z",
    "float": 222.5,
    "floatInStr": "22332.5",
    "int": 190,
    "intInStr": "23",
    "string": "a"
  },
  Object {
    "date": "2016-10-22T23:00:00.000Z",
    "float": 2.6,
    "floatInStr": "2.8",
    "int": 0,
    "intInStr": "0",
    "string": ""
  },
  Object {
    "date": "2016-10-11T23:00:00.000Z",
    "float": 12.5,
    "floatInStr": "13232.5",
    "int": 333,
    "intInStr": "11111",
    "string": "zz"
  },
  Object {
    "date": "2016-10-01T23:00:00.000Z",
    "float": 26.5,
    "floatInStr": "26.5",
    "int": 22,
    "intInStr": "222",
    "string": "ZZ"
  },
  Object {
    "date": "2003-06-31T23:00:00.000Z",
    "float": 452.5,
    "floatInStr": "33232.5",
    "int": 190,
    "intInStr": "23333",
    "string": "a"
  },
  Object {
    "date": "1982-10-31T23:00:00.000Z",
    "float": 2.5,
    "floatInStr": "2.5",
    "int": 200,
    "intInStr": "211",
    "string": "ååå"
  },
  Object {},
  Object {},
  Object {
    "floatInStr": "2.2"
  }
]
`;

exports[`SortObject Sort float (floatInStr) in string value 1`] = `
Array [
  Object {},
  Object {},
  Object {
    "floatInStr": "2.2"
  },
  Object {
    "date": "1982-10-31T23:00:00.000Z",
    "float": 2.5,
    "floatInStr": "2.5",
    "int": 200,
    "intInStr": "211",
    "string": "ååå"
  },
  Object {
    "date": "2016-10-22T23:00:00.000Z",
    "float": 2.6,
    "floatInStr": "2.8",
    "int": 0,
    "intInStr": "0",
    "string": ""
  },
  Object {
    "date": "2016-10-01T23:00:00.000Z",
    "float": 26.5,
    "floatInStr": "26.5",
    "int": 22,
    "intInStr": "222",
    "string": "ZZ"
  },
  Object {
    "date": "2016-10-11T23:00:00.000Z",
    "float": 12.5,
    "floatInStr": "13232.5",
    "int": 333,
    "intInStr": "11111",
    "string": "zz"
  },
  Object {
    "date": "2016-11-31T23:00:00.000Z",
    "float": 222.5,
    "floatInStr": "22332.5",
    "int": 190,
    "intInStr": "23",
    "string": "a"
  },
  Object {
    "date": "2003-06-31T23:00:00.000Z",
    "float": 452.5,
    "floatInStr": "33232.5",
    "int": 190,
    "intInStr": "23333",
    "string": "a"
  }
]
`;

exports[`SortObject Sort float (floatInStr) in string value descending 1`] = `
Array [
  Object {
    "date": "2003-06-31T23:00:00.000Z",
    "float": 452.5,
    "floatInStr": "33232.5",
    "int": 190,
    "intInStr": "23333",
    "string": "a"
  },
  Object {
    "date": "2016-11-31T23:00:00.000Z",
    "float": 222.5,
    "floatInStr": "22332.5",
    "int": 190,
    "intInStr": "23",
    "string": "a"
  },
  Object {
    "date": "2016-10-11T23:00:00.000Z",
    "float": 12.5,
    "floatInStr": "13232.5",
    "int": 333,
    "intInStr": "11111",
    "string": "zz"
  },
  Object {
    "date": "2016-10-01T23:00:00.000Z",
    "float": 26.5,
    "floatInStr": "26.5",
    "int": 22,
    "intInStr": "222",
    "string": "ZZ"
  },
  Object {
    "date": "2016-10-22T23:00:00.000Z",
    "float": 2.6,
    "floatInStr": "2.8",
    "int": 0,
    "intInStr": "0",
    "string": ""
  },
  Object {
    "date": "1982-10-31T23:00:00.000Z",
    "float": 2.5,
    "floatInStr": "2.5",
    "int": 200,
    "intInStr": "211",
    "string": "ååå"
  },
  Object {
    "floatInStr": "2.2"
  },
  Object {},
  Object {}
]
`;

exports[`SortObject Sort float in string value 1`] = `
Array [
  Object {},
  Object {},
  Object {
    "floatInStr": "2.2"
  },
  Object {
    "date": "1982-10-31T23:00:00.000Z",
    "float": 2.5,
    "floatInStr": "2.5",
    "int": 200,
    "intInStr": "211",
    "string": "ååå"
  },
  Object {
    "date": "2016-10-22T23:00:00.000Z",
    "float": 2.6,
    "floatInStr": "2.8",
    "int": 0,
    "intInStr": "0",
    "string": ""
  },
  Object {
    "date": "2016-10-11T23:00:00.000Z",
    "float": 12.5,
    "floatInStr": "13232.5",
    "int": 333,
    "intInStr": "11111",
    "string": "zz"
  },
  Object {
    "date": "2016-10-01T23:00:00.000Z",
    "float": 26.5,
    "floatInStr": "26.5",
    "int": 22,
    "intInStr": "222",
    "string": "ZZ"
  },
  Object {
    "date": "2016-11-31T23:00:00.000Z",
    "float": 222.5,
    "floatInStr": "22332.5",
    "int": 190,
    "intInStr": "23",
    "string": "a"
  },
  Object {
    "date": "2003-06-31T23:00:00.000Z",
    "float": 452.5,
    "floatInStr": "33232.5",
    "int": 190,
    "intInStr": "23333",
    "string": "a"
  }
]
`;

exports[`SortObject Sort float in string value descending 1`] = `
Array [
  Object {
    "date": "2003-06-31T23:00:00.000Z",
    "float": 452.5,
    "floatInStr": "33232.5",
    "int": 190,
    "intInStr": "23333",
    "string": "a"
  },
  Object {
    "date": "2016-11-31T23:00:00.000Z",
    "float": 222.5,
    "floatInStr": "22332.5",
    "int": 190,
    "intInStr": "23",
    "string": "a"
  },
  Object {
    "date": "2016-10-01T23:00:00.000Z",
    "float": 26.5,
    "floatInStr": "26.5",
    "int": 22,
    "intInStr": "222",
    "string": "ZZ"
  },
  Object {
    "date": "2016-10-11T23:00:00.000Z",
    "float": 12.5,
    "floatInStr": "13232.5",
    "int": 333,
    "intInStr": "11111",
    "string": "zz"
  },
  Object {
    "date": "2016-10-22T23:00:00.000Z",
    "float": 2.6,
    "floatInStr": "2.8",
    "int": 0,
    "intInStr": "0",
    "string": ""
  },
  Object {
    "date": "1982-10-31T23:00:00.000Z",
    "float": 2.5,
    "floatInStr": "2.5",
    "int": 200,
    "intInStr": "211",
    "string": "ååå"
  },
  Object {},
  Object {},
  Object {
    "floatInStr": "2.2"
  }
]
`;

exports[`SortObject Sort inputJsonObject on displayName 1`] = `
Array [
  Object {
    "displayName": "Sommerfugl",
    "id": 3,
    "identifier": Object {
      "museumNo": "C333",
      "subNo": "38"
    }
  },
  Object {
    "displayName": "Sverd",
    "id": 2,
    "identifier": Object {
      "museumNo": "C222",
      "subNo": "31"
    }
  },
  Object {
    "displayName": "Øks",
    "id": 5,
    "identifier": Object {
      "museumNo": "C666",
      "subNo": "3"
    }
  }
]
`;

exports[`SortObject Sort inputJsonObject on id 1`] = `
Array [
  Object {
    "displayName": "Sverd",
    "id": 2,
    "identifier": Object {
      "museumNo": "C222",
      "subNo": "31"
    }
  },
  Object {
    "displayName": "Sommerfugl",
    "id": 3,
    "identifier": Object {
      "museumNo": "C333",
      "subNo": "38"
    }
  },
  Object {
    "displayName": "Øks",
    "id": 5,
    "identifier": Object {
      "museumNo": "C666",
      "subNo": "3"
    }
  }
]
`;

exports[`SortObject Sort inputJsonObject on identifier 1`] = `
Array [
  Object {
    "displayName": "Øks",
    "id": 5,
    "identifier": Object {
      "museumNo": "C666",
      "subNo": "3"
    }
  },
  Object {
    "displayName": "Sverd",
    "id": 2,
    "identifier": Object {
      "museumNo": "C222",
      "subNo": "31"
    }
  },
  Object {
    "displayName": "Sommerfugl",
    "id": 3,
    "identifier": Object {
      "museumNo": "C333",
      "subNo": "38"
    }
  }
]
`;

exports[`SortObject Sort inputJsonObject on museumNo 1`] = `
Array [
  Object {
    "displayName": "Øks",
    "id": 5,
    "identifier": Object {
      "museumNo": "C666",
      "subNo": "3"
    }
  },
  Object {
    "displayName": "Sverd",
    "id": 2,
    "identifier": Object {
      "museumNo": "C222",
      "subNo": "31"
    }
  },
  Object {
    "displayName": "Sommerfugl",
    "id": 3,
    "identifier": Object {
      "museumNo": "C333",
      "subNo": "38"
    }
  }
]
`;

exports[`SortObject Sort inputJsonObject on subNo 1`] = `
Array [
  Object {
    "displayName": "Øks",
    "id": 5,
    "identifier": Object {
      "museumNo": "C666",
      "subNo": "3"
    }
  },
  Object {
    "displayName": "Sverd",
    "id": 2,
    "identifier": Object {
      "museumNo": "C222",
      "subNo": "31"
    }
  },
  Object {
    "displayName": "Sommerfugl",
    "id": 3,
    "identifier": Object {
      "museumNo": "C333",
      "subNo": "38"
    }
  }
]
`;

exports[`SortObject Sort int (intInStr) in string value 1`] = `
Array [
  Object {
    "date": "2016-10-22T23:00:00.000Z",
    "float": 2.6,
    "floatInStr": "2.8",
    "int": 0,
    "intInStr": "0",
    "string": ""
  },
  Object {},
  Object {},
  Object {
    "floatInStr": "2.2"
  },
  Object {
    "date": "2016-11-31T23:00:00.000Z",
    "float": 222.5,
    "floatInStr": "22332.5",
    "int": 190,
    "intInStr": "23",
    "string": "a"
  },
  Object {
    "date": "1982-10-31T23:00:00.000Z",
    "float": 2.5,
    "floatInStr": "2.5",
    "int": 200,
    "intInStr": "211",
    "string": "ååå"
  },
  Object {
    "date": "2016-10-01T23:00:00.000Z",
    "float": 26.5,
    "floatInStr": "26.5",
    "int": 22,
    "intInStr": "222",
    "string": "ZZ"
  },
  Object {
    "date": "2016-10-11T23:00:00.000Z",
    "float": 12.5,
    "floatInStr": "13232.5",
    "int": 333,
    "intInStr": "11111",
    "string": "zz"
  },
  Object {
    "date": "2003-06-31T23:00:00.000Z",
    "float": 452.5,
    "floatInStr": "33232.5",
    "int": 190,
    "intInStr": "23333",
    "string": "a"
  }
]
`;

exports[`SortObject Sort int (intInStr) in string value descending 1`] = `
Array [
  Object {
    "date": "2003-06-31T23:00:00.000Z",
    "float": 452.5,
    "floatInStr": "33232.5",
    "int": 190,
    "intInStr": "23333",
    "string": "a"
  },
  Object {
    "date": "2016-10-11T23:00:00.000Z",
    "float": 12.5,
    "floatInStr": "13232.5",
    "int": 333,
    "intInStr": "11111",
    "string": "zz"
  },
  Object {
    "date": "2016-10-01T23:00:00.000Z",
    "float": 26.5,
    "floatInStr": "26.5",
    "int": 22,
    "intInStr": "222",
    "string": "ZZ"
  },
  Object {
    "date": "1982-10-31T23:00:00.000Z",
    "float": 2.5,
    "floatInStr": "2.5",
    "int": 200,
    "intInStr": "211",
    "string": "ååå"
  },
  Object {
    "date": "2016-11-31T23:00:00.000Z",
    "float": 222.5,
    "floatInStr": "22332.5",
    "int": 190,
    "intInStr": "23",
    "string": "a"
  },
  Object {
    "date": "2016-10-22T23:00:00.000Z",
    "float": 2.6,
    "floatInStr": "2.8",
    "int": 0,
    "intInStr": "0",
    "string": ""
  },
  Object {},
  Object {},
  Object {
    "floatInStr": "2.2"
  }
]
`;

exports[`SortObject Sort int value 1`] = `
Array [
  Object {
    "date": "2016-10-22T23:00:00.000Z",
    "float": 2.6,
    "floatInStr": "2.8",
    "int": 0,
    "intInStr": "0",
    "string": ""
  },
  Object {},
  Object {},
  Object {
    "floatInStr": "2.2"
  },
  Object {
    "date": "2016-10-01T23:00:00.000Z",
    "float": 26.5,
    "floatInStr": "26.5",
    "int": 22,
    "intInStr": "222",
    "string": "ZZ"
  },
  Object {
    "date": "2016-11-31T23:00:00.000Z",
    "float": 222.5,
    "floatInStr": "22332.5",
    "int": 190,
    "intInStr": "23",
    "string": "a"
  },
  Object {
    "date": "2003-06-31T23:00:00.000Z",
    "float": 452.5,
    "floatInStr": "33232.5",
    "int": 190,
    "intInStr": "23333",
    "string": "a"
  },
  Object {
    "date": "1982-10-31T23:00:00.000Z",
    "float": 2.5,
    "floatInStr": "2.5",
    "int": 200,
    "intInStr": "211",
    "string": "ååå"
  },
  Object {
    "date": "2016-10-11T23:00:00.000Z",
    "float": 12.5,
    "floatInStr": "13232.5",
    "int": 333,
    "intInStr": "11111",
    "string": "zz"
  }
]
`;

exports[`SortObject Sort int value descending 1`] = `
Array [
  Object {
    "date": "2016-10-11T23:00:00.000Z",
    "float": 12.5,
    "floatInStr": "13232.5",
    "int": 333,
    "intInStr": "11111",
    "string": "zz"
  },
  Object {
    "date": "1982-10-31T23:00:00.000Z",
    "float": 2.5,
    "floatInStr": "2.5",
    "int": 200,
    "intInStr": "211",
    "string": "ååå"
  },
  Object {
    "date": "2016-11-31T23:00:00.000Z",
    "float": 222.5,
    "floatInStr": "22332.5",
    "int": 190,
    "intInStr": "23",
    "string": "a"
  },
  Object {
    "date": "2003-06-31T23:00:00.000Z",
    "float": 452.5,
    "floatInStr": "33232.5",
    "int": 190,
    "intInStr": "23333",
    "string": "a"
  },
  Object {
    "date": "2016-10-01T23:00:00.000Z",
    "float": 26.5,
    "floatInStr": "26.5",
    "int": 22,
    "intInStr": "222",
    "string": "ZZ"
  },
  Object {
    "date": "2016-10-22T23:00:00.000Z",
    "float": 2.6,
    "floatInStr": "2.8",
    "int": 0,
    "intInStr": "0",
    "string": ""
  },
  Object {},
  Object {},
  Object {
    "floatInStr": "2.2"
  }
]
`;

exports[`SortObject Sort string value 1`] = `
Array [
  Object {
    "date": "2016-10-22T23:00:00.000Z",
    "float": 2.6,
    "floatInStr": "2.8",
    "int": 0,
    "intInStr": "0",
    "string": ""
  },
  Object {},
  Object {},
  Object {
    "floatInStr": "2.2"
  },
  Object {
    "date": "2016-11-31T23:00:00.000Z",
    "float": 222.5,
    "floatInStr": "22332.5",
    "int": 190,
    "intInStr": "23",
    "string": "a"
  },
  Object {
    "date": "2003-06-31T23:00:00.000Z",
    "float": 452.5,
    "floatInStr": "33232.5",
    "int": 190,
    "intInStr": "23333",
    "string": "a"
  },
  Object {
    "date": "2016-10-01T23:00:00.000Z",
    "float": 26.5,
    "floatInStr": "26.5",
    "int": 22,
    "intInStr": "222",
    "string": "ZZ"
  },
  Object {
    "date": "2016-10-11T23:00:00.000Z",
    "float": 12.5,
    "floatInStr": "13232.5",
    "int": 333,
    "intInStr": "11111",
    "string": "zz"
  },
  Object {
    "date": "1982-10-31T23:00:00.000Z",
    "float": 2.5,
    "floatInStr": "2.5",
    "int": 200,
    "intInStr": "211",
    "string": "ååå"
  }
]
`;

exports[`SortObject Sort string value descending 1`] = `
Array [
  Object {
    "date": "1982-10-31T23:00:00.000Z",
    "float": 2.5,
    "floatInStr": "2.5",
    "int": 200,
    "intInStr": "211",
    "string": "ååå"
  },
  Object {
    "date": "2016-10-01T23:00:00.000Z",
    "float": 26.5,
    "floatInStr": "26.5",
    "int": 22,
    "intInStr": "222",
    "string": "ZZ"
  },
  Object {
    "date": "2016-10-11T23:00:00.000Z",
    "float": 12.5,
    "floatInStr": "13232.5",
    "int": 333,
    "intInStr": "11111",
    "string": "zz"
  },
  Object {
    "date": "2016-11-31T23:00:00.000Z",
    "float": 222.5,
    "floatInStr": "22332.5",
    "int": 190,
    "intInStr": "23",
    "string": "a"
  },
  Object {
    "date": "2003-06-31T23:00:00.000Z",
    "float": 452.5,
    "floatInStr": "33232.5",
    "int": 190,
    "intInStr": "23333",
    "string": "a"
  },
  Object {
    "date": "2016-10-22T23:00:00.000Z",
    "float": 2.6,
    "floatInStr": "2.8",
    "int": 0,
    "intInStr": "0",
    "string": ""
  },
  Object {},
  Object {},
  Object {
    "floatInStr": "2.2"
  }
]
`;

exports[`parseISODateNonStrict should accept full iso timestamp 1`] = `"2016-09-07T22:56:00.000Z"`;
